var/spack/repos/builtin/packages/grass/package.py

```diff
@@ -0,0 +1,176 @@
+##############################################################################
+#
+# This is a template package file for Spack.  We've put "FIXME"
+# next to all the things you'll want to change. Once you've handled
+# them, you can save this file and test your package like this:
+#
+#     spack install grass
+#
+# You can edit this file again by typing:
+#
+#     spack edit grass
+#
+# See the Spack documentation for more information on packaging.
+# If you submit this package back to Spack as a pull request,
+# please first remove this boilerplate and all FIXME comments.
+#
+from spack import *
+
+
+class Grass(AutotoolsPackage):
+    """GRASS GIS (Geographic Resources Analysis Support System), is a free
+       and open source Geographic Information System (GIS) software suite
+       used for geospatial data management and analysis, image processing,
+       graphics and maps production, spatial modeling, and visualization."""
+
+    homepage = "http://grass.osgeo.org"
+    url      = "https://grass.osgeo.org/grass74/source/grass-7.4.1.tar.gz"
+
+    version('7.4.1',    'bf7add62cbeb05a3ed5ad832344ba524')
+
+    variant('cxx',       default=True,  description='Add c++ functionality')
+    variant('tiff',      default=True,  description='Add TIFF functionality')
+    variant('png',       default=True,  description='Add PNG functionality')
+    variant('postgres',  default=False, description='Add PostgreSQL functionality')
+    variant('mysql',     default=False, description='Add MySQL functionality')
+    variant('sqlite',    default=True,  description='Add SQLite functionality')
+    variant('opengl',    default=True,  description='Add OpenGL functionality')
+    variant('fftw',      default=True,  description='Add FFTW functionality')
+    variant('blas',      default=False, description='Add BLAS functionality')
+    variant('lapack',    default=False, description='Add LAPACK functionality')
+    variant('cairo',     default=True,  description='Add Cairo functionality')
+    variant('freetype',  default=True,  description='Add FreeType functionality')
+    variant('readline',  default=False, description='Add Readline functionality')
+    variant('regex',     default=True,  description='Add regex functionality')
+    variant('pthread',   default=False, description='Add POSIX threads functionality')
+    variant('openmp',    default=False, description='Add OpenMP functionality')
+    variant('opencl',    default=False, description='Add OpenCL functionality')
+    variant('bzlib',     default=False, description='Add BZIP2 functionality')
+    variant('netcdf',    default=False, description='Enable NetCDF support')
+    variant('geos',      default=False, description='Geometry Engine for v.buffer')
+
+    # required components
+    depends_on('gmake@3.8.1:', type='build')
+    depends_on('zlib')
+    depends_on('flex')
+    depends_on('bison')
+    depends_on('proj')
+    depends_on('gdal')
```

Comment:
I was actually thinking about adding a GRASS package when I was working on GDAL. I noticed that GDAL has a `--with-grass` option, but that would create a circular dependency. Can you comment on whether GRASS should be built with GDAL or GDAL should be built with GRASS? I'm not too familiar with GRASS.

Reply:
This is a circular piece.  As I recall, you first build gdal without grass, then build grass, then build gdal with grass.  I could code this in as gdal~grass then have gdal to just code grass,  So gdal+grass should be able to compile then.

Reply:
Nah, I don't think that will work at the moment. Spack doesn't let the same package occur multiple times in the DAG, even if one is `gdal+grass` and the other is `gdal~grass`. I don't think it's possible to build GDAL with GRASS support at the moment.